WalletWasabi.Fluent/Models/Transactions/PrivacySuggestionsResult.cs

```diff
@@ -1,18 +1,76 @@
 using System.Collections.Generic;
-using System.Linq;
+using System.Threading.Tasks;
+using WalletWasabi.Fluent.Extensions;
 
 namespace WalletWasabi.Fluent.Models.Transactions;
 
-public record PrivacySuggestionsResult
+public record PrivacySuggestionsResult(IAsyncEnumerable<PrivacyItem> Items)
 {
-	public List<PrivacyWarning> Warnings { get; } = new();
-	public List<PrivacySuggestion> Suggestions { get; } = new();
+	private List<IAsyncEnumerable<PrivacyItem>> _combinedResults = new();
```

Comment:
nit, but more like a question... Can we somehow prioritize the items here? Because by the testing it seems to me during processing if the first item is a ChangeAvoidance item (and it takes seconds to finish), every other suggestion is waiting to be shown, despite their computation being already done.

We know that always the computation of the ChangeAvoidance items will take the most so put them at the end of the processing?

Reply:
@soosr I cannot reproduce this. Adding a 5 second delay in VerifyChangeAsync() line 218, I get all the other suggestions properly shown in the flyout and then the change-related ones appear in the end.

The `IAsyncEnumerable`s are added to a regular List<T> which is then iterated over, so we should get the results in the same order we insert them:

```csharp
result
					.Combine(VerifyLabels(info, transactionResult))
					.Combine(VerifyPrivacyLevel(info, transactionResult))
					.Combine(VerifyConsolidation(transactionResult))
					.Combine(VerifyUnconfirmedInputs(transactionResult))
					.Combine(VerifyCoinjoiningInputs(transactionResult))
					.Combine(VerifyChangeAsync(info, transactionResult, _linkedCancellationTokenSource));
```

Unless I'm missing something?

Reply:
I added a 5 seconds delay before this line
https://github.com/zkSNACKs/WalletWasabi/blob/aac6f76d745863c778090cafbbefa86b6b8ff48f/WalletWasabi.Fluent/Models/Transactions/PrivacySuggestionsModel.cs#L320

Just to simulate if the BnB algorithm runs longer (it can run up to 15 seconds).
After this, I did my testing by repeatedly creating a transaction where I know only the `Label Management` and a `Less` and a `More` change avoidance suggestion is possible.

I noticed that sometimes the `Label Management` option appears later. Taking a look at the logs I could see that when this scenario happened it always appeared after a change avoidance calculation was done.